sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -1,27 +1,30 @@
 package com.horizen.account.state
 
 import com.horizen.SidechainTypes
+import com.horizen.account.api.http.ZenWeiConverter
 import com.horizen.account.storage.AccountStateMetadataStorageView
 import com.horizen.account.transaction.EthereumTransaction
 import com.horizen.block.{MainchainBlockReferenceData, WithdrawalEpochCertificate}
+import com.horizen.box.data.WithdrawalRequestBoxData
 import com.horizen.box.{ForgerBox, WithdrawalRequestBox}
-import com.horizen.consensus.{ConsensusEpochInfo, ConsensusEpochNumber}
+import com.horizen.consensus.ConsensusEpochNumber
 import com.horizen.evm.{StateDB, StateStorageStrategy}
 import com.horizen.state.StateView
-import com.horizen.utils.{BlockFeeInfo, MerkleTree, WithdrawalEpochInfo}
+import com.horizen.utils.{BlockFeeInfo, WithdrawalEpochInfo}
 import scorex.core.VersionTag
-import scorex.util.{ModifierId, bytesToId}
 
 import java.math.BigInteger
 import scala.util.Try
 
-class AccountStateView( val metadataStorageView: AccountStateMetadataStorageView,
-                        val stateDb: StateDB,
-                        messageProcessors: Seq[MessageProcessor]) extends StateView[SidechainTypes#SCAT, AccountStateView] with AccountStateReader {
+class AccountStateView(val metadataStorageView: AccountStateMetadataStorageView,
+                       val stateDb: StateDB,
+                       messageProcessors: Seq[MessageProcessor]) extends StateView[SidechainTypes#SCAT, AccountStateView] with AccountStateReader {
   view: AccountStateView =>
 
   override type NVCT = this.type
 
+  lazy val withdrawalReqProvider: WithdrawalRequestProvider = messageProcessors.find(_.isInstanceOf[WithdrawalRequestProvider]).get.asInstanceOf[WithdrawalRequestProvider]
```

Comment:
why not just return `WithdrawalMsgProcessor` object?
Seems safe

Reply:
I wanted to keep the possibility to "inject" the withdrawalmesgprocessor from outside, using the messageProcessors list.
Using directly WithdrawalMsgProcessor object would have removed this possibility.

Reply:
Good. Then no observations.